### PR TITLE
Feature 192 change project enabled state

### LINKED ISSUE #192: Admin shall be able to change project access level
## Situation
At the moment a project cannot be deactivated or set to read only state. To prevent access to a project, users must be removed from project. This is not very suitable.

The administrator shall be able to set an access level for a project.

- FULL_ACCESS: no restrictions at all (intial default)
- READ_ONLY: only read access is possible, no changes
- NO_ACCCESS: users have no access to reports or jobs any longer  
  _(but administrators will still have possiblitity to access project information by special administration REST API)_

## Solution
- Administration domain
  - REST access
  - store access level inside project entity, migration script sets `FULL_ACCESS` to all existing projects
  - trigger async `PROJECT_ACCESS_LEVEL_CHANGED` message

- Scheduler domain
  - introduce `SchedulerProjectConfig` and set project access level there
  - assert project access levels
    - if a `SchedulerProjectConfig` exists check the access level is accepted for the action
    - if no `SchedulerProjectConfig` exists accept as `FULL_ACCESS`
    - all scheduler REST actions are blocked except getJobStatus - letter one is accepted when project access level is `READ_ONLY` but not for `NO_ACCESS`
  - handle message  `PROJECT_ACCESS_LEVEL_CHANGED`, store `SchedulerProjectConfig`

- Scan domain
  - reuse `ScanProjectConfig`, introduce new ScanProjectConfigID `PROJECT_ACCESS_LEVEL` and set project access level there
  - assert project access levels
    - if a `ScanProjectConfig` exists for key `PROJECT_ACCESS_LEVEL` and project, check the access level is accepted for the action
    - if no `ScanProjectConfig` exists for project and key `PROJECT_ACCESS_LEVEL` accept as `FULL_ACCESS`
    - all scan REST actions are blocked when `NO_ACCCESS`. `READ_ONLY` is always accepted (because user have only read operations inside this domain )
  - handle message  `PROJECT_ACCESS_LEVEL_CHANGED`, store /create `SchedulerProjectConfig`

- Notification domain
  - handle  `PROJECT_ACCESS_LEVEL_CHANGED` : inform administrators about project access level change

## Conststraints

When a project has access level `READ_ONLY`
- a new scan is not possible, HTTP 403: Forbidden will be sent for:
  - create job
  - approve existing job
  - upload job data
- running jobs are still running
- already scheduled jobs will be scheduled
- user are still able to check status for a job in their project
- user can still download their former reports

When a project has access level `NO_ACCESS`
- a new scan is not possible, HTTP 403: Forbidden will be sent for:
  - create job
  - approve existing job
  - upload job data
- running jobs are still running
- already scheduled jobs will be scheduled
- user are NOT able to check status for a job in their project,  HTTP 403: Forbidden will be sent
- user can NOT download their former reports,  HTTP 403: Forbidden will be sent

## Addtional
- Documentation
- Change usecase `UseCaseAdministratorShowsProjectDetails`: 
  - add project access level to project information inside output
  - change existing RestDoc test, so the information is also inside generated OpenAPI
 - Delete project workflow
   - domain schedule must drop SchedulerProjectConfig entries

### LINKED ISSUE #216: Fix spring security context problem on event bus execution
### Situation
ProductResultService contained a  `@RolesAllowed` annotation with super admin role constraint

A rest controller which was also secured to super admin role and did call some services
which leads to execution of product result service via Event Bus.

### Problem
The Spring security context is not accepted :
Here an output from integration test UC_ADMIN_RESTARTS_JOB_HARD__simulate_accidently_job_restarted_where_already_done

log:
```
...
2020-04-23 20:02:55.903 DEBUG 14188 --- [xecutor_thread3] c.d.s.domain.scan.ScanMessageHandler     : received synchron domain request: DomainMessage [id=REQUEST_PURGE_JOB_RESULTS, parameters={environment.base.url=https://localhost:8443, sechub.uuid=4b9eaff1-a118-4f7d-b44f-1c053969cf5e}]
2020-04-23 20:02:55.905 ERROR 14188 --- [xecutor_thread3] c.d.s.domain.scan.ScanMessageHandler     : Was not able to purge results for job {}

org.springframework.security.authentication.AuthenticationCredentialsNotFoundException: An Authentication object was not found in the SecurityContext
	at org.springframework.security.access.intercept.AbstractSecurityInterceptor.credentialsNotFound(AbstractSecurityInterceptor.java:379)
	at org.springframework.security.access.intercept.AbstractSecurityInterceptor.beforeInvocation(AbstractSecurityInterceptor.java:223)
	at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:65)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at com.daimler.sechub.domain.scan.product.ProductResultService$$EnhancerBySpringCGLIB$$d1ad7bdd.deleteAllResultsForJob(<generated>)
	at com.daimler.sechub.domain.scan.ScanMessageHandler.handleJobRestartHardRequested(ScanMessageHandler.java:106)
	at com.daimler.sechub.domain.scan.ScanMessageHandler.receiveSynchronMessage(ScanMessageHandler.java:94)
	at com.daimler.sechub.sharedkernel.messaging.DomainMessageService.sendSynchron(DomainMessageService.java:147)
	at com.daimler.sechub.sharedkernel.messaging.DomainMessageService$$FastClassBySpringCGLIB$$c7de0c21.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:685)
	at com.daimler.sechub.sharedkernel.messaging.DomainMessageService$$EnhancerBySpringCGLIB$$83df6b4.sendSynchron(<generated>)
	at com.daimler.sechub.domain.schedule.SchedulerRestartJobService.sendPurgeJobResultsRequest(SchedulerRestartJobService.java:202)
	at com.daimler.sechub.domain.schedule.SchedulerRestartJobService.restartJob(SchedulerRestartJobService.java:120)
	at com.daimler.sechub.domain.schedule.SchedulerRestartJobService.restartJobHard(SchedulerRestartJobService.java:84)
	at com.daimler.sechub.domain.schedule.ScheduleMessageHandler.handleJobRestartHardRequested(ScheduleMessageHandler.java:119)
	at com.daimler.sechub.domain.schedule.ScheduleMessageHandler.receiveAsyncMessage(ScheduleMessageHandler.java:99)
	at com.daimler.sechub.sharedkernel.messaging.DomainMessageService$AsynchronMessageHandlerTaskAdapter.run(DomainMessageService.java:199)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

2020-04-23 20:02:55.906 ERROR 14188 --- [xecutor_thread3] c.d.s.s.messaging.DomainMessageService   : Was not able to run request:DomainMessage [id=REQUEST_JOB_RESTART_HARD, parameters={environment.base.url=https://localhost:8443, job.restart.data={"jobUUID":"4b9eaff1-a118-4f7d-b44f-1c053969cf5e"}}] with handler com.daimler.sechub.domain.schedule.ScheduleMessageHandler@73852720
...
```

### Workaround
Temporarily commented out the annotation inside service  - but solution necessary!

### Solution
Move security context into call
e.g. something like:

 ```
 try {
            SecurityContext ctx = SecurityContextHolder.createEmptyContext();
            SecurityContextHolder.setContext(ctx);
            ctx.setAuthentication(event.getAuthentication());

            //Do what ever you want to do

        } finally {
            SecurityContextHolder.clearContext();
        }
 ```

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #192 
- closes #216

### CODE REVIEW

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: sechub-administration/src/main/java/com/daimler/sechub/domain/administration/project/ProjectTransactionService.java [hunk 1]
```diff
@@ -40,6 +40,12 @@ public Project saveInOwnTransaction(Project project) {
 		return result;
 	}
 
+	/**
+	 * Persists a project and a user entity in same transaction
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-administration/src/main/java/com/daimler/sechub/domain/administration/project/ProjectTransactionService.java:44]: done

File: sechub-administration/src/test/java/com/daimler/sechub/domain/administration/project/ProjectChangeAccessLevelServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        serviceToTest.changeProjectAccessLevel(PROJECT1_ID, ProjectAccessLevel.FULL);
+
+        /* test */
+        verify(eventBus,never()).sendAsynchron(any());
+        verify(eventBus,never()).sendSynchron(any());
+    }
+    
+    
+    @Test
+    void an_change_from_full_access_to_read_only_does_trigger_one_async_event_containing_old_and_new_access_level() {
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-administration/src/test/java/com/daimler/sechub/domain/administration/project/ProjectChangeAccessLevelServiceTest.java:68]: done

File: sechub-administration/src/main/java/com/daimler/sechub/domain/administration/project/ProjectChangeAccessLevelService.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Autowired
+    AuditLogService auditLogService;
+
+    /* @formatter:off */
+    @UseCaseAdministratorChangesProjectAccessLevel(
+			@Step(
+					number = 2,
+					name = "Change access level",
+					description = "The service will change the access level and trigger event"))
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-administration/src/main/java/com/daimler/sechub/domain/administration/project/ProjectChangeAccessLevelService.java:56]: Done: Changed to
"The service will change the project access level inside administration domain and trigger a change event to inform recipients about the new situation."

File: sechub-doc/src/docs/asciidoc/documents/code2doc/usecases/admin/changeProjectAccessLevel.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+An administrator is able to changes project access level. 
+Project access levels do some constraints to possible user operations for
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/documents/code2doc/usecases/admin/changeProjectAccessLevel.adoc:3]: done: changed to:

> An administrator is able to change the access level of a project. 
> 
> Project access levels do constraint possible user operations for 
> a project - e.g. restrict to read only operations. The details of different 
> restrictions are described below.

File: sechub-doc/src/test/java/com/daimler/sechub/restdoc/ProjectAdministrationRestControllerRestDocTest.java [hunk 5]
```diff
@@ -369,20 +419,21 @@ public void restdoc_show_project_details() throws Exception {
...
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_WHITELIST).description("A list of all whitelisted URIs. Only these ones can be scanned for the project!"),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_METADATA).description("An JSON object containing metadata key-value pairs defined for this project."),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_METADATA + ".key1").description("An arbitrary metadata key"),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_DESCRIPTION).description("The project description.")
+				responseFields(
+                            fieldWithPath(ProjectDetailInformation.PROPERTY_PROJECT_ID).description("The name of the project"),
+                            fieldWithPath(ProjectDetailInformation.PROPERTY_USERS).description("A list of all users having access to the project"),
+							fieldWithPath(ProjectDetailInformation.PROPERTY_OWNER).description("Username of the owner ofthis project. An owner is the person in charge."),
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/test/java/com/daimler/sechub/restdoc/ProjectAdministrationRestControllerRestDocTest.java:428]: done twice - same typo did exist in another method of same file as well.

File: sechub-scan/src/test/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+import com.daimler.sechub.sharedkernel.project.ProjectAccessLevel;
+import com.daimler.sechub.sharedkernel.validation.ProjectIdValidation;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ScanProjectConfigAccessLevelServiceTest {
+
+    private static final String PROJECT1 = "project1";
+    private ScanProjectConfigAccessLevelService serviceToTest;
+    private ProjectIdValidation projectIdValidation;
+    private ScanProjectConfigService scanprojectConfigService;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-scan/src/test/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelServiceTest.java:19]: done

File: sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigService.java [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        }
+
+        config.setProjectAccessLevel(newAccessLevel);
+
+        repository.save(config);
+    }
+
+    /**
+     * Resolve the access level for the given project. If no project configuration exists, it will be
+     * automatically created. Created project configuration do always have full access level enabled.
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigService.java:56]: I will not add the "do" : When using translation software like google translate and retranslate english->german->english the do is always removed. So it looks like this is not correct.
But will apply the plural form:

> Created project configurations always have the full access level activated.

File: sechub-doc/src/test/java/com/daimler/sechub/restdoc/ProjectAdministrationRestControllerRestDocTest.java [hunk 8]
```diff
@@ -444,25 +496,26 @@ public void restdoc_change_project_description() throws Exception {
...
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_WHITELIST).description("A list of all whitelisted URIs. Only these ones can be scanned for the project!"),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_METADATA).description("An JSON object containing metadata key-value pairs defined for this project."),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_METADATA + ".key1").description("An arbitrary metadata key."),
-                                    fieldWithPath(ProjectDetailInformation.PROPERTY_DESCRIPTION).description("The project description.")
+                responseFields(
+                            fieldWithPath(ProjectDetailInformation.PROPERTY_PROJECT_ID).description("The name of the project."),
+                            fieldWithPath(ProjectDetailInformation.PROPERTY_USERS).description("A list of all users having access to the project."),
+                            fieldWithPath(ProjectDetailInformation.PROPERTY_OWNER).description("Username of the owner ofthis project. An owner is the person in charge."),
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/test/java/com/daimler/sechub/restdoc/ProjectAdministrationRestControllerRestDocTest.java:505]: done

File: sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+    private static final ScanProjectConfigID CONFIG_ID_PROJECT_ACCESS_LEVEL = ScanProjectConfigID.PROJECT_ACCESS_LEVEL;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ScanProjectConfigAccessLevelService.class);
+
+    @Autowired
+    ProjectIdValidation projectIdValidation;
+
+    @Autowired
+    ScanProjectConfigService scanprojectConfigService;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java:29]: done

File: sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+    public void changeProjectAccessLevel(String projectId, ProjectAccessLevel newAccessLevel, ProjectAccessLevel formerAccessLevel) {
+        /* validate */
+        notNull(newAccessLevel, "New accesslevel may not be null!");
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java:33]: done

File: sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigService.java [hunk 11]
```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.domain.schedule.config;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigService.java:2]: done

File: sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.domain.schedule.config;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java:2]: done

File: sechub-shared-kernel/src/main/java/com/daimler/sechub/sharedkernel/messaging/ProjectMessage.java [hunk 13]
```diff
@@ -7,6 +7,7 @@
 
 import com.daimler.sechub.commons.model.JSONable;
 import com.daimler.sechub.sharedkernel.MustBeKeptStable;
+import com.daimler.sechub.sharedkernel.project.ProjectAccessLevel;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-shared-kernel/src/main/java/com/daimler/sechub/sharedkernel/messaging/ProjectMessage.java:10]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AssertProject.java [hunk 14]
```diff
@@ -211,4 +212,16 @@ public AssertProject hasMetaData(Map<String, String> expectedMap) {
         return this;
     }
 
+    public AssertProject hasAccessLevel(ProjectAccessLevel none) {
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AssertProject.java:215]: changed to "level". Also added a a validation that tests cannot use `null` inside.

File: sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+        scanprojectConfigService.set(projectId, CONFIG_ID_PROJECT_ACCESS_LEVEL, newAccessLevel.getId());
+
+        LOG.info("Changed access level for project:{} to level:{}", projectId, newAccessLevel.getId());
+
+    }
+
+    /**
+     * Fetches project access level for given project - if no project access level
+     * is defined, default will be used, which is
```
Comment (Jeeppler) [inline sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java:56]: Where is the rest of the sentence? What is the default?

File: sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+        config.projectAccessLevel = ProjectAccessLevel.FULL;
+
+        when(repository.findById(PROJECT1)).thenReturn(Optional.of(config));
+
+        /* execute */
+        serviceToTest.changeProjectAccessLevel(PROJECT1, ProjectAccessLevel.READ_ONLY, null);
+
+        /* test */
+        verify(repository).save(config);
+        verify(assertService).asserProjectIdValid(PROJECT1);
```
Comment (Jeeppler) [inline sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java:70]: I think, a `t` is missing in the method name `asserProjectIdValid`

```suggestion
        verify(assertService).assertProjectIdValid(PROJECT1);
```

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java (hunk 15 above)
Comment (de-jcup) [reply sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java:56]: changed to:
Fetches project access level for given project - if no project access level
is defined, default will be used, which is {@value #DEFAULT_ACCESS_LEVEL}

File: sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java (hunk 16 above)
Comment (de-jcup) [reply sechub-schedule/src/test/java/com/daimler/sechub/domain/schedule/config/SchedulerProjectConfigServiceTest.java:70]: done

**Comment by de-jcup** [comment]
There were also some wrong usages of Objects.nonNull(...) - where it was used as an assert method - nonNull does only return a boolean but does not assert. So these usages must be replaced by Objects.requireNonNull(...)

Will be fixed with next review commit as well

File: sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (Jeeppler) [reply sechub-scan/src/main/java/com/daimler/sechub/domain/scan/project/ScanProjectConfigAccessLevelService.java:29]: No, it is not changed, therefore unresolved it.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Are there any plans to add the functionality described in #192 to the Admin UI? Because, the Admin UI does not have any menu item to change the project access level.

**Comment by de-jcup** [comment]
With last commit there is a new action inside DAUI available to change project access level. After project selection there appears a dialog which shows up current access level and the possible changes:

image

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java [hunk 17]
```diff
@@ -0,0 +1,23 @@
+package com.daimler.sechub.integrationtest.api;
```
Comment (Jeeppler) [inline sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java:1]: Please apply the formatter.

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.daimler.sechub.integrationtest.internal.TestScenario;
+
+public class FixedTestProject extends TestProject{
```
Comment (Jeeppler) [inline sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java:5]: I still have no idea what this class is suppose to do or used for. I would prefer a JavaDoc comment describing the purpose of the class.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments

File: sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/ui/AbstractListDialogUI.java [hunk 19]
```diff
@@ -16,6 +16,7 @@
 import javax.swing.Action;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/ui/AbstractListDialogUI.java:16]: I created https://github.com/Daimler/sechub/issues/771 to handle this in future more automated. It is very cumbersome to have those (but valid) findings inside reviews.

[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/ui/AbstractListDialogUI.java:16]: done

File: sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/DeveloperAdministration.java [hunk 20]
```diff
@@ -729,4 +739,8 @@ public void removeProjectIdsFromProfile(String profileId, List<String> list) {
         removeProjectIdsFromProfile(profileId, list.toArray(new String[list.size()]));
     }
 
+    public void changeProjectAccessLevel(String projectId, ProjectAccessLevel accessLevel) {
+        asTestUser().changeProjectAccessLevel(new FixedTestProject(projectId), accessLevel);
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/DeveloperAdministration.java:743]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java (hunk 17 above)
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java:1]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java (hunk 18 above)
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java:5]: already done - see my former comment . I added javadoc for FixedTestProject and also for FixedTestuser

Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/FixedTestProject.java:5]: done

File: sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/ui/action/project/ChangeProjectAccessLevelAction.java [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+
+        ChangeProjectAccessLevelDialogUI dialogUI = new ChangeProjectAccessLevelDialogUI(getContext(), "Change access level for project:" + projectId,
+                accesslevel);
+        dialogUI.setDescription("Current access level for project:" + accesslevel);
+        dialogUI.showDialog();
+
+        if (!dialogUI.isOkPressed()) {
+            return;
+        }
+        ProjectAccessLevel accessLevel = dialogUI.getSelectedValue();
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-developertools/src/main/java/com/daimler/sechub/developertools/admin/ui/action/project/ChangeProjectAccessLevelAction.java:42]: changed to: currentAccessLevel and wantedAccessLevel so the meaning should be clear now.

**Review by Jeeppler — APPROVED** [review]
